### PR TITLE
Fix for admonitions being displayed out of their original context.

### PR DESCRIPTION
**Pull request by intelkevinputnam**
Removes 'pullup' for admonitions in detailed description to keep them in context.

File changed:  breathe/renderer/sphinxrenderer.py

This is a fix for #889

### CODE REVIEW

**Comment by vermeeren** [comment]
(Sort of replies to https://github.com/breathe-doc/breathe/issues/889#issuecomment-1526689401)

@intelkevinputnam Via blame I see the commit that added this is https://github.com/breathe-doc/breathe/commit/7f579b132c1212f8dd79ba160d831df7898f9b83, PR #670. Quite long ago but rationale seems to be that it matched Sphinx's own style.

As a result, I'm not sure about potential side effects this might have for other users. @jakobandersen Do you remember or have insight is whether `pullup` is needed for these cases?

If it was only for stylistic reason with the default big block rendering of `note` and `warning` (PR #670 screenshot), it might be best to turn this into a configuration option as there are likely users that prefer `pullup` style of rendering.

**Comment by intelkevinputnam** [comment]
Sounds good. I'll update the PR as a configurable option. Any guidance on naming the parameter?

**Comment by vermeeren** [comment]
@intelkevinputnam Initially I'm thinking something like `breathe_detaileddesc_pullup_types` which one can set to desired types from https://pydoc.dev/docutils/latest/docutils.nodes.html. I'm hoping this extra flexibility will help with some other requests made over the years about changing the order, as you can manually input everything in a very specific order if desired.

```python
# user in conf.py will have to import
# (unless there is a better way, but this errors early, instead of at runtime)
from docutils import nodes

# this would match current behaviour.
breathe_detaileddesc_pullup_types = [ nodes.note, nodes.warning ]
```

Then the implementation can loop over the supplied types and pull them up.

As for the default value, I think empty `[]` is good, as I agree this is kind of an unexpected ordering change when using Breathe. We can document in the `documentation/source/directives.rst` that this is `since v4.36.0` and to get `v4.35.0 or older` behaviour one can set it to above snippet. That should help with ctrl+f on version numbers when this change is released.

Thanks for the help!

**Comment by intelkevinputnam** [comment]
@vermeeren 

Gotcha. Easily doable. 

Some additional thoughts: 

TL;DR - would it make sense to define different behaviors for `doxygenpage` vs `doxygengroup` or `doxygenclass`?

I ran into this problem because I am dealing with pages and pages of documentation written in Markdown with admonitions scattered throughout. The presentation is meant to be linear for all the content which is all contained in the `detaileddescription` field of the Doxygen XML. I use the `doxygenpage` directive to render them.

Based on the original PR, I think this feature was developed for classes (for example) where there are many objects that might be rendered in any order. This would help to collect them in a sensible way. 

Which might mean that a given user might want one behavior for pages and another for groups and classes. What do you think?

**Comment by vermeeren** [comment]
@intelkevinputnam Very good catch, makes sense to do it based upon directive then. We could update the setting value to use an array per directive in a dictionary instead of a plain array in this case.

```python
# no entry for "doxygenpage", so would default to []
breathe_detaileddesc_pullup_types = {
	"doxygenclass": [ nodes.note, nodes.warning ],
	"doxygengroup": []
}
```

Not 100% sure if above syntax is correct, but the general idea should be good. I am not sure if we should set one or more directives to not-`[]` by default however, I still sort of feel that Breathe shouldn't modify (sort) input data too much if it wasn't asked to. Your thoughts on this?

**Comment by intelkevinputnam** [comment]
@vermeeren I agree that sorting should be a requested behavior. I'll take a stab at implementing it.

**Comment by intelkevinputnam** [comment]
@vermeeren I used strings to denote the node to pull up rather than the type value.

`breathe_detaileddesc_pullup_types = {"doxygenpage":["note","warning","fieldlist"]}`

**Comment by intelkevinputnam** [comment]
@vermeeren can you help with this check error? I tried to match `self.context.directive_args` usage in other places in `sphinxrenderer.py`, but the linter doesn't seem to like it.

**Comment by intelkevinputnam** [comment]
@michaeljones It looks to me like the linter is now blocking my PR on a file I haven't changed. Should I go ahead and submit the correction it is asking for?

**Comment by intelkevinputnam** [comment]
@michaeljones It looks to me like the linter is now blocking my PR on a file I haven't changed. Should I go ahead and submit the correction it is asking for?

**Comment by anjalirajasekhar** [comment]
Hi, I had a similar issue where the notes where not displayed in order. I tried the solution provided by removing the pullups. But still the issue was not resolved. I was wondering is there any other reasons which might lead to this behaviour.

**Comment by anjalirajasekhar** [comment]
Hi @intelkevinputnam. I had a similar issue where the notes where not displayed in order. I tried removing the pullups. But still the behavior is same. Is there some other reasons that might lead to this kind of behavior

**Comment by intelkevinputnam** [comment]
@anjalirajasekhar I just returned from vacation, so I'm a little rusty. Which directive are you using?
